pinecone/core/openapi/shared/api_client.py

```diff
@@ -8,6 +8,24 @@
 import typing
 from urllib.parse import quote
 from urllib3.fields import RequestField
+import time
+import random
+
+def retry_api_call(
+    func, args=(), kwargs={}, retries=3, backoff=1, jitter=0.5
+):
+    attempts = 0
+    while attempts < retries:
+        try:
+            return func(*args, **kwargs)  # Attempt to call __call_api
```

Comment:
I'm not a python expert, but isn;t there some library that provides a decorator to retry functions? seems like this should be a solved problem

Reply:
There's nothing in the standard library afaik, and I'm very hesitant to add third-party dependencies because of the overall dependency hell situation in python.